internal/controller/deployment_controller.go

```diff
@@ -87,6 +89,63 @@ func (r *DeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return r.Update(ctx, l, deployment)
 }
 
+func (r *DeploymentReconciler) setStatusFromClusterStatus(ctx context.Context, l logr.Logger, deployment *hmc.Deployment) (bool, error) {
+	dc, err := dynamic.NewForConfig(r.Config)
+	if err != nil {
+		return true, fmt.Errorf("failed to create client configuration: %v", err)
+	}
+
+	resourceId := schema.GroupVersionResource{
+		Group:    "cluster.x-k8s.io",
+		Version:  "v1beta1",
+		Resource: "clusters",
+	}
+
+	list, err := dc.Resource(resourceId).Namespace(deployment.Namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
```

Comment:
I think it's more reliable to use helm labels instead of a specific name as naming patterns are not enforced. I think it's fine to assume that we expect exactly one cluster object for now.

Reply:
Is the concern around uniqueness? I believe the cluster name must be unique within a namespace.

Reply:
No guarantee that cluster.Name == deployment.Name

Reply:
Ah ok I've changed it to use the label